java/ql/lib/semmle/code/java/Expr.qll

```diff
@@ -717,6 +718,19 @@ class DoubleLiteral extends Literal, @doubleliteral {
 /** A character literal. For example, `'\n'`. */
 class CharacterLiteral extends Literal, @characterliteral {
   override string getAPrimaryQlClass() { result = "CharacterLiteral" }
+
+  /**
+   * Gets a string which consists of the single character represented by
+   * this literal.
+   */
+  string getStringValue() { result = getValue() }
```

Comment:
Why add this alias?

Reply:
Mostly because `getValue()` just happens to represent the same result (but not necessarily on purpose) and this separate predicate makes the intention and behavior clearer.
Though maybe it indeed does not add much value, should I remove it?

However, should it then (in a separate issue?) also be discussed if [`StringLiteral.getRepresentedString()`](https://codeql.github.com/codeql-standard-libraries/java/semmle/code/java/Expr.qll/predicate.Expr$StringLiteral$getRepresentedString.0.html) is really needed? Because there `getValue()` would be even more obvious to use.

Reply:
We should probably just get rid of `getRepresentedString` through a proper deprecation cycle. The reason for its existence dates back to before we had proper extraction of literal values, when this would simply remove the leading and trailing `"` chars in a naive attempt to calculate the value from the literal in QL.